release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+          MergeReleaseNotes.class.getSimpleName());
+    }
+
+    MergeReleaseNotes mergeReleaseNotes = new MergeReleaseNotes();
+    mergeReleaseNotes.createMergedReleaseNote(args);
+  }
+
+  public void createMergedReleaseNote(String... args) throws Exception {
+    for (String path : args) {
+      Edition edition;
+      Repository repository;
+
+      switch (path) {
+        case "scalardb.md":
+          edition = Edition.COMMUNITY;
+          repository = Repository.DB;
+          break;
+        case "cluster.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.CLUSTER;
+          break;
+        case "graphql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.GRAPHQL;
+          break;
+        case "sql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.SQL;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown file name: " + path);
+      }
+
+      File file = new File(path);
+      load(edition, repository, file);
+    }
+    output();
+  }
+
+  void load(Edition edition, Repository repository, File file) throws Exception {
+    Category category = null;
+    String line;
+    Matcher matcher;
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+      while ((line = reader.readLine()) != null) {
+        if (DEBUG != null) System.err.printf("ReadLine: %s%n", line);
+        matcher = Pattern.compile("^## *(\\p{Print}+) *$").matcher(line);
+        if (matcher.matches()) {
+          if (DEBUG != null) System.err.printf("matched: %s%n", matcher.group(1));
+          if (!SECTION_SUMMARY.equalsIgnoreCase(matcher.group(1)))
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
+            throw new IllegalStateException(
+                "Missing category. Release note text: " + matcher.group(1));
+          addReleaseNote(new ReleaseNote(edition, category, repository, matcher.group(1)));
+        }
+      }
+    }
+  }
+
+  private void addReleaseNote(ReleaseNote releaseNote) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap =
+        editionMap.computeIfAbsent(releaseNote.edition, k -> new EnumMap<>(Category.class));
+
+    Map<Repository, List<ReleaseNote>> repositoryMap =
+        categoryMap.computeIfAbsent(releaseNote.category, k -> new EnumMap<>(Repository.class));
+
+    List<ReleaseNote> releaseNotesList =
+        repositoryMap.computeIfAbsent(releaseNote.repository, k -> new ArrayList<>());
+
+    if (releaseNote.edition.equals(Edition.ENTERPRISE)) {
+      Matcher matcher =
+          Pattern.compile("(.*) +(\\((#[0-9]+ *)+\\))$").matcher(releaseNote.releaseNoteText);
```

Comment:
> assigning a compiled pattern to a variable with a meaningful name so that the readers can easily understand what to match.

I agree. I'll revise it to have more meaningful name variable.

Reply:
Done.  26e04ab